pkg/subctl/cmd/gather/cni.go

```diff
@@ -107,6 +121,43 @@ func logIPTablesCmds(info Info, pod *v1.Pod) {
 	}
 }
 
+func OVNData(info Info) {
+	if networkPluginCNIType[info.Submariner.Status.NetworkPlugin] != typeOvn {
+		return
+	}
+
+	pods, err := findPods(info.ClientSet, ovnMasterPodLabelOCP)
+	if err != nil && apiErrors.IsNotFound(err) {
+		pods, err = findPods(info.ClientSet, ovnMasterPodLabelGeneric)
+		if err != nil {
+			info.Status.QueueFailureMessage("Failed to gather any OVN master pods: " + err.Error())
+		}
+	}
+
+	var masterPod *v1.Pod
+	// ovn-nbctl commands only work on one of the masters, figure out which one
+	for i := range pods.Items {
+		err = logCmdOutput(info, &pods.Items[i], ovnShowCmd, ovnShowName)
+		if err == nil {
+			masterPod = &pods.Items[i]
+		}
+	}
+
+	if masterPod == nil {
+		info.Status.QueueFailureMessage(fmt.Sprintf("Failed to exec OVN command in all masters: %s", err))
+		return
+	}
+
+	for name, command := range ovnCmds {
+		err = logCmdOutput(info, masterPod, command, name)
+		if err != nil {
+			info.Status.QueueFailureMessage(fmt.Sprintf("Failed to exec OVN command: %s", err))
+		}
+	}
+
+	info.Status.QueueSuccessMessage("Successfully gathered OVN details")
```

Comment:
Probably we can do something like 

```
	if status.HasFailureMessages() {
		status.End(cli.Failure)
		return
	}
        info.Status.QueueSuccessMessage("Successfully gathered OVN details")
        status.End(cli.Success)
```

Reply:
I don't think we want to end the status since it's shared. Also there could be failure messages from prior components. It might be better for each component to create its own status but that would require a refactor. I suggest following what we've done for other components, ie log what it will be doing but no specific success message at the end. Any subsequent failure messages indicate it failed.

So above L151:

`info.Status.QueueSuccessMessage(fmt.Sprintf("Gathering OVN data from master pod %", masterPod.Name))`